src/core/server/http/base_path_proxy_server.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { RouteOptions, Server } from 'hapi-latest';
```

Comment:
I started noticing this naming--i wonder if it's better to name this hapi-17, not hapi-latest. That's quickly out of date.

Reply:
Or do you see us constantly updating the hapi-latest to be 18, 19, etc?

Reply:
> Or do you see us constantly updating the hapi-latest to be 18, 19, etc?

Yeah, that was my idea, I really don't want to get into sitatuation when we're several major versions behind for the _most important_ libs once again, it's a pain to update and we should eventually have a "defined process" around constantly keeping our dependencies up-to-date